Sources/PopupBridge/POPPopupBridge.swift

```diff
@@ -0,0 +1,194 @@
+import Foundation
+import SafariServices
+import WebKit
+
+@objcMembers
+public class POPPopupBridge: NSObject, WKScriptMessageHandler, SFSafariViewControllerDelegate {
+    
+    // MARK: - Private Properties
+    
+    private let messageHandlerName = "POPPopupBridge"
+    private let hostName = "popupbridgev1"
+    
+    private let webView: WKWebView
+    private let delegate: POPPopupBridgeDelegate
+    private var safariViewController: SFSafariViewController?
+    
+    private static var scheme: String?
+    private static var returnBlock: ((URL) -> Bool)?
+    
+    // MARK: - Initializer
+    
+    // TODO: - make unfailable
+    public init?(webView: WKWebView, delegate: POPPopupBridgeDelegate) {
+        // TODO: - require scheme in init, versus injecting via static method.
+        guard let scheme = POPPopupBridge.scheme else {
+            let exception = NSException(
+                name: NSExceptionName(rawValue: "POPPopupBridgeSchemeNotSet"),
+                reason: "PopupBridge requires a URL scheme to be set"
+            )
+            exception.raise()
+            return nil
+        }
+        
+        self.webView = webView
+        self.delegate = delegate
+        
+        super.init()
+        
+        configureWebView(scheme: scheme)
+        
+        // TODO: - Remove returnBlock definition from init
+        POPPopupBridge.returnBlock = { (url : URL) -> Bool in
+            guard let script = self.constructJavaScriptCompletionResult(returnURL: url) else {
+                return false
+            }
+            
+            self.injectWebView(webView: webView, withJavaScript: script)
+            return true
+        }
+    }
+    
+    // MARK: - Public Methods
+    
+    // TODO: - Remove this static method and require scheme in init
+    /// Set the URL Scheme that you have registered in your Info.plist.
+    public static func setReturnURLScheme(_ returnURLScheme: String?) {
+        self.scheme = returnURLScheme
+    }
+    
+    /// Handle completion of the popup flow by calling this method from either
+    /// your scene:openURLContexts: scene delegate method or
+    /// your application:openURL:sourceApplication:annotation: app delegate method.
+    @objc(openURL:)
+    public static func open(url: URL) -> Bool {
+        if let returnBlock {
+            return returnBlock(url)
+        } else {
+            return false
+        }
```

Comment:
```suggestion
    returnBlock?(url) ?? false
```
I'm not sure if this is valid, but we could leverage optional syntax more here. Even if this is valid, is the loss of expressiveness worth it? Brevity may decrease readability in this case.

Reply:
Good call - I have a [TODO](https://github.com/braintree/popup-bridge-ios/blob/swift-conversion/Sources/PopupBridge/POPPopupBridge.swift#L41) to remove this whole separate `returnBlock` concept in a follow-up PR. 

Gonna 🏈 punt on this feedback for that follow up - it'll also probably be obsolete in that follow up.